Feat: PostCreatePage 마크업 추가

wherehows opened the pull request:
## 이슈 번호 <!-- 관련된 이슈의 번호를 # 뒤에 입력해주세요  -->

- close: #39 

## 특이 사항
- Category별로 컴포넌트를 정의함. 디자인 시스템 관점에서, 기존에 작성했던 컴포넌트의 수준보다 한 단계 위인 컴포넌트를 구성해볼 수 있어 보임. 가령 Category별로 컴포넌트를 쪼개 놓긴 했으나, 하나의 Category 컴포넌트로 여러 개의 Category에서 재활용 가능하게.
하지만 지금은 Category 컴포넌트 재사용에 대한 고민보다는 단순히 React 로직 작성의 편의를 위한 목적으로 Category를 쪼개놓음. 추가적으로 본인이 디자인 시스템에 대한 개념이 부족하므로, 추후 공부한 뒤 리펙토링 하겠음.

- Category간 간격은 ThemeProvider에 등록 예정

wherehows committed:
Refactor: SelectionBox 컴포넌트 수정

- options.map에서 options?.map으로 코드 변경
- 마크업 편의를 위해option list의 key를 index로 사용
- label과의 grouping을 위해id Prop 추가

wherehows committed:
Refactor: Label 컴포넌트 수정

- color와 bgColor Prop hexCode가 아닌 컬러 변수명으로 받도록 설정

wherehows committed:
Refactor: constants 파일 수정

- GENDER 네임 스페이스 내에 UNKNOWN 프로퍼티 추가

wherehows committed:
Refactor: Input 컴포넌트 수정

- margin Prop 추가

wherehows committed:
Refactor: SelectionBox 컴포넌트 수정

- margin Prop 추가

wherehows committed:
Refactor: CheckBox 컴포넌트 수정

- 기본 font-size를 1rem에서 1.6rem으로 변경
- display flex에서 display inline-flex로 변경

wherehows committed:
Refactor: ThemeProvider에 normalOrange 컬러 추가

wherehows committed:
Refactor : Input 컴포넌트 수정

- margin Prop 추가

wherehows committed:
Feat: PostCreatePage 마크업 추가

Jihyeon228 committed:
Feat: ShortHeader 마크업 추가 (#32)

* Feat: Seperator 마크업 추가

* Feat: SidebarButton 마크업 추가

* Refactor: children defaultProps 삭제

* Refactor: 코드리뷰 반영
- React => {useState}로 사용
- margin 단축 속성으로 변경

* Feat: NotificationButton 마크업 추가

* Refactor: Avatar에 margin 추가

* Feat: ShortHeader 마크업 추가

sanoopark reviewed (approved):
비슷한 모양의 중복된 컴포넌트가 있기 때문에 추후 추상화가 가능할 것으로 보입니다. (논의완료)

wherehows committed:
Feat: LongHeader 마크업 추가 (#28)

* Feat: LongHeader 마크업 추가

* Refactor: Search 아이콘 컬러 Brand 컬러로 변경

* Refactor: Design 디테일 수정

- menu 버튼 컬러 brand 컬러로 변경
- 실종 및 보호와 보호소 동물 버튼에 아이콘 추가

* Refactor: Button 컴포넌트 수정

- fontWeight Prop 추가
- Button 컴포넌트 내 children 가운데 정렬

* Refactor: Button 그림자 normal에서 light로 변경

* Chore: 실종 및 보호와 보호소 동물 아이콘 추가

* Refactor: Login 여부에 따라 아이콘 렌더링 분기 로직 추가

* Refactor: Image 경로 가져오는 함수 helper 파일 내에 등록

* Refactor: 아이콘 컬러 브랜드 컬러로 변경

- Notification 및 Information 아이콘 brand 컬러로 변경

* Refactor: Image 컴포넌트 margin Prop 추가

* Refactor: Button에 전달되는 bgColor 값을  컬러명으로 변경

* Refactor: MiddleContainer 내 Icon과 글자간 거리 조정

* Refactor: Logo 컬러 브랜드 컬러로 변경

* Refactor: Input placeholder의 font-weight normal로 변경

* Refactor: styled된 Mui Icon 컴포넌트 명 변경

* Refactor: 돋보기 아이콘 수직정렬 방법 변경

* Refactor: Image 컴포넌트에 margin Prop 추가

swprk committed:
Feat: 회원가입 유효성 검사 (#43)

* Refactor: Input onChange, onBlur Props 추가

* Feat: FormError 마크업 추가

* Feat:  회원가입 유효성 검사

* Refactor: FormError Prop 네이밍 수정

* Chore: Formik 관련 누락분 추가

* Refactor: FormError Props isRequired 해제

wherehows committed:
Refactor: Label 컴포넌트 수정 (#46)

* Refactor: Label 컴포넌트 수정

- display:inline-flex를 이용한 글자 가운데 정렬
- ThemeProvider에 width 등록
- forHtml Prop isRequired 제거

* Refactor: Label 컴포넌트 수정

- color, bgColor Prop을 hexCode가 아닌 컬러 명으로 입력받도록 수정

* Refactor: ThemeProvider의 label 프로퍼티 수정

wherehows committed:
Feat: Slider 마크업 추가 (#37)

* Feat: Slider 마크업 추가

* Refactor: Button 컴포넌트 수정

- Props 추가(onClick, top, right, left, bottom, position)
- flex를 이용한 가운데 정렬

* Refactor: ThemeProvider에 slider 관련 size 추가

* Refactor: Slider 컴포넌트 수정

- Dots 컴포넌트에 전달하는 length Prop값을 imageList.length로 변경

* Chore: Conflict 해결

wherehows committed:
Refactor: Header Dots 수정

- Wrapper 태그 button에서 div로 변경
- Dot 태그 div에서 button으로 변경

wherehows committed:
Feat: Slider 추가

wherehows committed:
Chore: Conflict 해결
